app/main/views/agreements.py

```diff
@@ -16,6 +17,76 @@
 from ...supplier_utils import validate_agreement_details_data
 
 
+@main.route('/agreements', methods=['POST'])
+def create_framework_agreement():
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+    json_has_required_keys(json_payload, ["agreement"])
+    update_json = json_payload["agreement"]
+
+    json_has_keys(update_json, required_keys=['supplierId', 'frameworkSlug'])
+
+    supplier_framework = SupplierFramework.find_by_supplier_and_framework(
+        update_json['supplierId'], update_json['frameworkSlug']
+    )
+
+    if not supplier_framework or not supplier_framework.on_framework:
+        abort(
+            404,
+            "supplier_id '{}' is not on framework '{}'".format(
+                update_json['supplierId'], update_json['frameworkSlug']
+            )
+        )
+
+    # We enforce only one framework agreement per SupplierFramework for the moment
+    # This behaviour will be refactored out later to allow for multiple
+    existing_framework_agreement = FrameworkAgreement.query.filter(
+        FrameworkAgreement.supplier_id == update_json['supplierId']
+    ).filter(
+        Framework.slug == update_json['frameworkSlug']
+    ).first()
+
+    if existing_framework_agreement:
+        abort(
+            400,
+            "supplier_id '{}' already has a framework agreement for framework '{}'".format(
+                update_json['supplierId'], update_json['frameworkSlug']
+            )
+        )
+
+    framework = Framework.query.filter(
+        Framework.slug == update_json['frameworkSlug']
+    ).first_or_404()
+
+    framework_agreement = FrameworkAgreement(
+        supplier_id=update_json['supplierId'],
+        framework_id=framework.id
+    )
+
+    try:
+        db.session.add(framework_agreement)
+        db.session.flush()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, e.orig)
+
+    audit_event = AuditEvent(
+        audit_type=AuditTypes.create_agreement,
+        user=updater_json['updated_by'],
+        data={
+            'supplierId': update_json['supplierId'],
+            'frameworkSlug': update_json['frameworkSlug']
+        },
+        db_object=framework_agreement
+    )
+
+    db.session.add(audit_event)
+    db.session.commit()
+
+    return jsonify(agreement=framework_agreement.serialize()), 201
+
+
+# TODO Do we really need this route?
```

Comment:
Feels like we're more likely to need a route to get by supplier_id and framework_slug rather than directly by ID.  But let's leave it how it is until we work out what we actually need.